lib/spack/spack/schema/packages.py

```diff
@@ -41,6 +41,8 @@
                             "spec": {"type": "string"},
                             "message": {"type": "string"},
                             "when": {"type": "string"},
+                            "root_only": {"type": "boolean"},
```

Comment:
I don't know that `roots` is the right name here. The PR says this is for build-only packages.  What is the relation here between those and roots?